mlflow/tracking/fluent.py

```diff
@@ -72,25 +72,29 @@
 
 
 class DefaultSetContextVar:
-    def __init__(self, contextvar, default: Callable):
-        self.contextvar = contextvar
+    def __init__(self, name: str, default: Callable):
+        self.contextvar = contextvars.ContextVar(name)
         self.default = default
 
     def get(self):
         try:
             return self.contextvar.get()
-        except LookupError:
-            if self.default is not None:
+        except LookupError as e:
+            if self.default is None:
+                raise e
+            else:
```

Comment:
Do we need this if-else?

Reply:
I copied the `ContextVar` API where `default` is `Optional` (need to fix typing for `init` actually), in which case this check seemed necessary. But I'm also open to just making it required since otherwise you'd just use the standard contextvar class and not need this

Reply:
+1 for making it required to simplify the code. This is not a public class. We can make it optional whenever we want in the future.